bundle/src/main/java/com/adobe/acs/commons/http/impl/HttpClientFactoryImpl.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2013 - 2015 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+package com.adobe.acs.commons.http.impl;
+
+import com.adobe.acs.commons.http.HttpClientFactory;
+import org.apache.felix.scr.annotations.*;
+import org.apache.http.HttpHost;
+import org.apache.http.client.fluent.Executor;
+import org.apache.http.client.fluent.Request;
+import org.apache.sling.commons.osgi.PropertiesUtil;
+
+import java.util.Map;
+
+
+@Component(
+        label = "Http Components Fluent Executor Factory",
+        description = "Http Components Fluent Executor Factory", immediate =
+        false,
+        metatype = false, configurationFactory = true, policy =
```

Comment:
We dont want `metatype=true` ? Seems like might be handy for developers to use. Ultimately these configs should be deployed via osgiConfigs, but does it hurt to allow config via Felix?